pkg/extended_registry/aserto.go

```diff
@@ -0,0 +1,204 @@
+package extendedregistry
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aserto-dev/go-grpc/aserto/api/v1"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/tidwall/gjson"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type AsertoClient struct {
+	base *xClient
+}
+
+func NewAsertoClient(logger *zerolog.Logger, cfg *Config, client *http.Client) ExtendedClient {
+	baseClient := newExtendedClient(logger, cfg, client)
+
+	return &AsertoClient{
+		base: baseClient,
+	}
+}
+
+func (c *AsertoClient) ListOrgs() ([]string, error) {
+	address, err := c.extendedAPIAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	jsonBody, err := c.base.get(address + "/api/v1/registry/organizations")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list organizations")
+	}
+	type Org struct {
+		Name string `json:"name"`
+	}
+	// TODO Add paginated information to read all orgs
+	var parseStruct struct {
+		Orgs []Org `json:"orgs"`
+	}
+
+	err = json.Unmarshal([]byte(jsonBody), &parseStruct)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal orgs")
+	}
+	var response []string
+	for i := range parseStruct.Orgs {
+		response = append(response, parseStruct.Orgs[i].Name)
+	}
+	return response, nil
+}
+
+func (c *AsertoClient) ListRepos(org string) ([]*api.PolicyImage, error) {
+	if org == "" {
+		return nil, nil
+	}
+	address, err := c.extendedAPIAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	jsonBody, err := c.base.get(address + "/api/v1/registry/images")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list images")
+	}
+
+	response := struct {
```

Comment:
should we use: https://github.com/aserto-dev/go-grpc/blob/main/aserto/registry/v1/registry.pb.go#L65 ?